api-definitions/db_positions/api.yaml

```diff
@@ -0,0 +1,292 @@
+  ProblemJson:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v10.2.3/openapi/definitions.yaml#/ProblemJson"
+  PaginationResponse:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v10.2.3/openapi/definitions.yaml#/PaginationResponse"
+  PaymentAmount:
+    description: |-
+      Amount of payment in euro cent. PagoPA accepts up to 9999999999 euro cents.
+    type: integer
+    minimum: 1
+    maximum: 9999999999
+    example: 15000
+  Iban:
+    type: string
+    pattern: '[a-zA-Z]{2}[0-9]{2}[a-zA-Z0-9]{1,30}'
+    example: IT60X0542811101000000123456
+  DetailResponse:
+    type: object
+    properties:
+      detail:
+        type: string
+    required:
+      - detail
+  DepartmentName:
+    type: string
+    description: |-
+      The department inside the organization that runs the service. Will
+      be added to the content of sent messages.
+    minLength: 1
+    example: ufficio tributi comune di Roma
+  Taxonomy:
+    type: string
+    pattern: "^[0-9]{3}-[a-zA-Z]$"
+    example: 010-TARI/TEFA
+    description: Payment Tax see [Tabella - Tassonomia dei servizi di incasso](https://docs.google.com/spreadsheets/d/13xOd__Qd4pwKHr3wjE-73NAB2O7UKmIt/edit#gid=1444196601)
+  Transfer:
+    type: object
+    properties:
+      partial_amount:
+        $ref: "#/definitions/PaymentAmount"
+      transfer_IBAN:
+        $ref: "#/definitions/Iban"
+      pa_fiscal_code:
+        $ref: "#/definitions/OrganizationFiscalCode"
+      reason:
+          type: string
+          minLength: 1
+          example: Causale pagamento TARI comune di ROMA 
+      taxonomy:
+          $ref: "#/definitions/Taxonomy"
+    required:
+      - partial_amount
+      - transfer_IBAN
+      - fiscal_code
+      - taxonomy
+  Debtor:
+    type: object
+    properties:
+      name:
+        type: string
+        minLength: 1
+        example: Mario
+      surname:
+        type: string
+        minLength: 1
+        example: Rossi
+      fiscal_code:
+        $ref: "#/definitions/FiscalCode"
+      type:
+        type: string
+        enum:
+          - FISICA
+          - GIURIDICA
+        description: Payment Status  
+    required:
+      - name
+      - surname
+      - type
+  Payment-Option:
+    type: object
+    properties:
+      payment-option-id:
+        type: string
+        minLength: 1
+        example: 12345678901-009876543212346543-01
+      amount:
+        $ref: "#/definitions/PaymentAmount"
+      due_date:
+        $ref: "#/definitions/Timestamp"
+      is_conclusive:
+        type: boolean
+        default: true
+      payment_description: 
+        type: string
+        minLength: 1
+        example: TARI 2021 soluzione unica
+      transfers:
+        type: array
+        items:
+          $ref: "#/definitions/Transfer"
+    required:
+      - payment-option-id
+      - amount
+      - due_date
+      - is_conclusive
+      - description
+      - transfers
+  PaymentPosition:
```

Comment:
I would add a `version` field to keep track of changes, or else it'll be a nightmare to cache this data and keep track of changes

Reply:
@cloudify which usage do you imagine? I see version as an internal information, not something the api expects from and communicate to the consumer

Reply:
Say we keep a cache somewhere (e.g. for IO), we need a way to know if a position needs to be updated, you can compare the cached version against the latest version

Reply:
btw that's what we do for all mutable models in IO (e.g. service)

Reply:
isn't clear to me too. @cloudify  can give an example of the problem you are referring to

Reply:
@pasqualespica on IO we defined an abstraction over db models that, when updating a record, instead of changing the actual item it creates a brand new record, bumping a version number by one. Here the code: https://github.com/pagopa/io-functions-commons/blob/1a55096465b62b025a3111d02461d87f837411d9/src/utils/cosmosdb_model_versioned.ts#L73

To achieve that, model definitions are enriched with metadata such as `version`

Reply:
tnx @balanza, I'd add `npx oval validate -p api.yaml` check too, to the existing `github action`, as you talked about it with @BurnedMarshal . Do you think this is a good idea? I have seen it on `pagopa-proxy` pipelines too.

Reply:
alway a good Idea to add a check

Reply:
Just for me to understand, since this is the interface with the PA to let them inform us about the state of a Debtor Position, why the need of adding the version here? 
imho, for IO we will need (I suppose) a new GET api based only on the citizen Fiscal Code, in that api the version will be useful.

Reply:
Probably we'll need to add a form of versioning for the objects in next release of the API specification.